### ISSUE TITLE
Feature request: drag from legend

### ISSUE THREAD

**Issue by PeterJohnson** [epezent/implot#89]
My application has multiple visible plots.  It’s really easy to drag from a data source to a particular plot, but it would also be a nice feature to be able to drag a series from the legend of one plot to another plot.  API wise I'm thinking something like (between BeginPlot and EndPlot), mimicing ImGui::BeginDragDropSource():
```
if (ImPlot::BeginLegendDragDropSource("id")) {
  ImGui::SetDragDropPayload(...);
  // text etc for DND tooltip
  ImPlot::EndLegendDragDropSource();
}
```

I think a temporary workaround would be to use the legend hovered feature and detect drags, but that feels very manual and error-prone.

**Comment by epezent** [comment]
Interesting idea! If you can submit a working prototype that wraps the current API (e.g. using legend hovered API as you suggested) and/or minimal changes to the inner workings, I'd be happy to consider it.

**Comment by ozlb** [comment]
> Interesting idea! If you can submit a working prototype that wraps the current API (e.g. using legend hovered API as you suggested) and/or minimal changes to the inner workings, I'd be happy to consider it.

@epezent 
I think that is missing also a mechanism (function SetNext___ or something) in order to temporary disable axis drag&drop without locking. This is a must if you want to drag&drop something that has been rendered after BeginPlot. What do you think, what is your idea about?

**Comment by PeterJohnson** [comment]
I have this working and will PR it soon.  It was pretty straightforward to do based on the ImGui::BeginDragDropSource() implementation.  It's somewhat magical.  It's not necessary to disable axis drag and drop for it to work.

**Comment by epezent** [comment]
Excellent! Can't wait to see the PRs

**Comment by PeterJohnson** [comment]
Closed by #106.
